### PR TITLE
Copy slice along arbitrary axis

### PR DESCRIPTION
**Pull request by pluskid**
Added this to allow data-parallelization when the first dimension is not batch size.

### CODE REVIEW

File: include/mxnet/c_api.h [hunk 1]
```diff
@@ -321,10 +321,31 @@ MXNET_DLL int MXNDArraySlice(NDArrayHandle handle,
...
+ * \return 0 when success, -1 when failure happens
+ *
+ * Unlike the MXNDArraySlice method in which the sliced
+ * array shares memory with the original array. This
+ * method always copies the data.
+ *
+ * Also this method expects that out is pre-allocated with
+ * appropriate shape.
+ */
+MXNET_DLL int MXNDArrayCopySliceTo(NDArrayHandle handle,
```
Comment (tqchen) [inline include/mxnet/c_api.h]: I think this should not be part of C API, but an operator that does this job

Comment (tqchen) [inline include/mxnet/c_api.h]: see how copyto is redirected into an ndarray operator https://github.com/dmlc/mxnet/blob/master/python/mxnet/ndarray.py#L472

**Comment by piiswrong** [comment]
I think you can use mx.nd.crop for this

**Comment by pluskid** [comment]
i need to copy into pre allocated ndarrays and avoid making extra copies of memory. for data par, the targets are all pre allocated.

**Comment by piiswrong** [comment]
crop(ou=target) should do it

**Comment by pluskid** [comment]
hmm, this should do the trick. I missed this one. Thanks!

**Pull request merged**